process/block/displayBlock.go

```diff
@@ -110,7 +82,7 @@ func (txc *transactionCounter) displayLogInfo(
 	arguments := []interface{}{
 		"total txs processed", txc.totalTxs,
 		"block txs processed", txc.currentBlockTxs,
-		"txs in pool", txc.getNumTxsFromPool(selfId, dataPool, numShards),
+		"txs in pool", txc.getTxPoolCounts(dataPool).String(),
```

Comment:
Possible race condition here, as getTxPoolCounts returns the reference to a shardedcounts object, and here in the String method you access the internal map without protecting it with the same mutex as for the PutCounts.

Reply:
`PutCounts` supposed to be invoked only by the owner of the counts object (the one who constructs it) - and therefore the `PutCounts` isn't listed in the interface. I will make it concurrent safe though.